core/dictgen/src/rootcling_impl.cxx

```diff
@@ -772,9 +772,17 @@ string gLibsNeeded;
 
 ////////////////////////////////////////////////////////////////////////////////
 
-void RecordDeclCallback(const char *c)
+void RecordDeclCallback(const clang::RecordDecl* recordDecl)
 {
-   string need(gAutoloads[c]);
+   std::string need;
+   if (recordDecl->hasOwningModule()) {
+      clang::Module *M = recordDecl->getOwningModule()->getTopLevelModule();
+      need = "lib" + M->Name + ".so";
+   } else {
+      auto N = clang::dyn_cast<const clang::NamedDecl> (recordDecl);
+      need = gAutoloads[N->getName()];
```

Comment:
You do not need to cast recordDecl to a NamedDecl, RecordDecl derives from NamedDecl.

Instead of `N->getName()` we seem to use `N->getNameForDiagnostic(stream,N->getASTContext().getPrintingPolicy(),true);`